thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/classification/classifier/AnomalyClassifierFactory.java

```diff
@@ -0,0 +1,133 @@
+          configPath, e);
+    }
+  }
+
+  /**
+   * Instantiates a factory that has the configuration file, which is given as an input stream, to external classes.
+   */
+  public AnomalyClassifierFactory(InputStream input) {
+    loadPropertiesFromInputStream(input);
+  }
+
+  /**
+   * Loads the configuration file to external classes.
+   *
+   * @param input the input steam of the configuration file.
+   */
+  private void loadPropertiesFromInputStream(InputStream input) {
+    try {
+      props.load(input);
+    } catch (IOException e) {
+      LOG.error("Error loading the alert filters from config", e);
+    } finally {
+      IOUtils.closeQuietly(input);
+    }
+  }
+
+  /**
+   * Given a spec, which is a map of string to a string, of the anomaly classifier, returns a anomaly classifier
+   * instance. The implementation of the instance could be located in an external package or ThirdEye project.
+   *
+   * If the type of provider exists in both the external package and ThirdEye project, then the implementation from
+   * external package will be used.
+   *
+   * @param spec a map of string to a string.
+   *
+   * @return a anomaly classifier instance.
+   */
+  public AnomalyClassifier fromSpec(Map<String, String> spec) {
+    if (spec == null) {
+      spec = Collections.emptyMap();
+    }
+    // the default recipient provider is a dummy recipient provider
+    AnomalyClassifier anomalyClassifier = DUMMY_ANOMALY_CLASSIFIER;
+    if (spec.containsKey(ANOMALY_CLASSIFIER_TYPE_KEY)) {
+      String recipientProviderType = spec.get(ANOMALY_CLASSIFIER_TYPE_KEY);
+      // We first check if the implementation (class) of this provider comes from external packages
+      if(props.containsKey(recipientProviderType.toUpperCase())) {
+        String className = props.getProperty(recipientProviderType.toUpperCase());
+        try {
+          anomalyClassifier = (AnomalyClassifier) Class.forName(className).newInstance();
+        } catch (Exception e) {
+          LOG.warn(e.getMessage());
+        }
+      } else { // otherwise, we try to look for the implementation from internal classes
+        anomalyClassifier = fromStringType(spec.get(ANOMALY_CLASSIFIER_TYPE_KEY));
+      }
+    }
+    anomalyClassifier.setParameters(spec);
+
+    return anomalyClassifier;
+  }
+
+  /**
+   * The available classifier type in ThirdEye project.
+   */
+  public enum AnomalyClassifierType {
+    DUMMY
+  }
+
+  /**
+   * The methods returns the instance of anomaly classifier whose implementation is located in ThirdEye project.
+   *
+   * @param type the type name of the provider.
+   *
+   * @return a instance of anomaly classifier.
+   */
+  private static AnomalyClassifier fromStringType(String type) {
+    if (StringUtils.isBlank(type)) { // safety check and backward compatibility
+      return DUMMY_ANOMALY_CLASSIFIER;
+    }
+
+    AnomalyClassifierType providerType = AnomalyClassifierType.DUMMY;
```

Comment:
can we call is classifierType everywhere in this file instead of providerType? What is the meaning of providerType?